wallet/wallet-cli/src/repl/non_interactive/mod.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod log;
+
+use tokio::sync::mpsc;
+
+use crate::{
+    cli_event_loop::Event, cli_println, commands::ConsoleCommand, console::ConsoleContext,
+    errors::WalletCliError,
+};
+
+use super::{get_repl_command, parse_input};
+
+pub fn run<T: std::io::BufRead>(
+    input: std::io::Lines<T>,
+    output: &ConsoleContext,
+    event_tx: mpsc::UnboundedSender<Event>,
+) -> Result<(), WalletCliError> {
+    let repl_command = get_repl_command();
+
+    for line in input {
+        match line {
+            Ok(line) => {
+                let res = parse_input(&line, &repl_command);
+                match res {
+                    Ok(Some(command)) => {
+                        let res = super::run_command_blocking(&event_tx, command);
+
+                        match res {
+                            Ok(cmd) => match cmd {
+                                ConsoleCommand::Print(text) => {
+                                    cli_println!(output, "{}", text);
+                                }
+                                ConsoleCommand::ClearScreen => {
+                                    cli_println!(output, "Not supported");
```

Comment:
I would break and return an error when using unsupported commands in non-interactive mode.

Reply:
I think breaking on unsupported commands should be part of more robust error handling. I updated wallet-cli to break on a first error in non-interactive mode (can be also enabled/disabled from the command line).

Reply:
Yep. In non-interactive mode should break and exit. But in interactive mode it should just print the error. Though I'm making this comment because this file is in non-interactive.